ops/testing.py

```diff
@@ -1883,7 +1884,8 @@ def run_action(self, action_name: str,
         action_under_test = _RunningAction(action_name, ActionOutput([], {}), params)
         handler = getattr(self.charm.on, f"{action_name.replace('-', '_')}_action")
         self._backend._running_action = action_under_test
-        handler.emit()
+        self._action_id_counter += 1
+        handler.emit(self._action_id_counter)
```

Comment:
```suggestion
        handler.emit(str(self._action_id_counter))
```

Also would probably be good to have a test that catches this.